### PR TITLE
Fix GraphEdit port connection when graph is zoomed/unzoomed

### LINKED ISSUE #46425: Can't grab outputs from visual shader node's when the visual shader editor is zoomed out a little
**Godot version:**

3.2.3 stable

**OS/device including version:**

Windows 10, with a 3200x1800 display at 250% scaling if that's relevant

**Issue description:**

https://user-images.githubusercontent.com/50304111/109214998-c67ba800-776f-11eb-8d65-724ed8947837.mp4

**Steps to reproduce:**
Try what I tried in the video

**Minimal reproduction project:**

### PR DESCRIPTION
**Pull request by Chaosus**
test

Fix #46425

### CODE REVIEW

File: scene/gui/graph_edit.cpp [hunk 1]
```diff
@@ -760,6 +760,11 @@ bool GraphEdit::is_in_hot_zone(const Vector2 &pos, const Vector2 &p_mouse_pos) {
 			continue;
 		}
 		Rect2 rect = child->get_rect();
+
+		// To prevent intersections with other nodes.
+		rect.set_position(rect.position * zoom);
+		rect.set_size(rect.size * zoom);
```
Comment (KoBeWi) [inline scene/gui/graph_edit.cpp:766]: You could just do
```
Rect2 rect = child->get_rect() * zoom;
```

Comment (Chaosus) [reply scene/gui/graph_edit.cpp:766]: That would be cool but Rect2 does not have such operator :/

Comment (KoBeWi) [reply scene/gui/graph_edit.cpp:766]: Ah, right. But
```
rect.position *= zoom
rect.size *= zoom
```
should be possible.

Comment (Chaosus) [reply scene/gui/graph_edit.cpp]: Fixed.

**Review by KoBeWi — APPROVED** [review]

**Comment by akien-mga** [comment]
I tried to cherry-pick for 3.2.4 but aside from usual style conflicts, it seems there might be a previous change to GraphEdit which was not cherry-picked and is used here (`connecting_valid`), so this might be worth backporting with a dedicated PR (maybe including that other previous commit if it's also safe for 3.2).

**Comment by akien-mga** [comment]
Will be cherry-picked via #47173.

**Pull request merged**